grade.go

```diff
@@ -72,6 +72,78 @@ func compare(expected, actual string) (bool, string) {
 	return evaluateDiff(d), d
 }
 
+func handleDirectives(expected, actual []string) (fixedExpected, fixedActual []string) {
+	// If there is no bang-sh
+	if !strings.Contains(expected[0], "!#") {
+		return expected, actual
+	} else {
+		fixedExpected = expected[1:]
+		fixedActual = actual
+		if strings.Contains(expected[0], "c") {
+			for i := range fixedExpected {
+				// set all lowercase
+				fixedExpected[i] = strings.ToLower(fixedExpected[i])
+			}
+			for i := range fixedActual {
+				fixedActual[i] = strings.ToLower(fixedActual[i])
+			}
+		}
+
+		if strings.Contains(expected[0], "w") {
+			for i := range fixedExpected {
+				// set all lowercase
+				fixedExpected[i] = strings.ReplaceAll(fixedExpected[i], " ", "")
+			}
+			for i := range fixedActual {
+				fixedActual[i] = strings.ReplaceAll(fixedActual[i], " ", "")
+			}
+		}
+	}
+	return fixedExpected, fixedActual
+}
+
+// Function that evaluates student program output by computing it to expected output
+// Supports custom syntax in out.txt file, represented by the Syntax Dictionary in support.go
+func processOutput(expected, actual string) []string {
+
+	SyntaxDictionary := initSyntaxDictionary()
+
+	// Convert strings into array of strings separated by a newline and manipulate text to handle any directives
+	expectedLines, actualLines := handleDirectives(strings.Split(expected, "\n"), strings.Split(actual, "\n"))
+
+	// Variable to track position in actualLines[]
+	position := 0
+
+	// Integer array containing evaluation of each line. Values either 1 or 0.
+	results := make([]string, len(expectedLines))
+
+	// Loop across each line of expected to compare to actual
+	for i, line := range expectedLines {
+		if i > 0 {
+			position++ // Increment each step after first pass
+		}
+		// if there are no more lines of actual output to compare it to, break
+		if i+1 > len(actualLines) {
+			break
+		}
+
+		// See if line starts with special character indicating use of custom syntax
+		if strings.HasPrefix(line, "!") {
```

Comment:
also an opportunity for a helper method- something like

```go
package opengrader.directives

type directive int

const (
  MENU directive = iota
  IGNORE directive = iota
  NONE directive = iota
)

func getDirective(line string) (directive)
```

...or something similar. that way we can farm off the logic to the getDirective function and not worry about replicating it elsewhere.